From f3f400a0d974b3e6d268ce494f38e3aa3dc1ac2a Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Fri, 21 Jul 2017 14:59:05 +0800
Subject: [PATCH 1172/1345] ARM64: pinctrl: marvell: armada-37xx: update mpp
 definition to A0 version

commit  53e1aa1e6337a86f49c3070d8a2082f8a1199e4c from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Current Armada-37xx mpp definition is from Z1 version, this patch
updates mpp definition to A0 version:
- A new group "smi" is added in A0 with 2 pins - mpp2[19:18];
- Group "pcie1" in A0 has 3 pins - mpp2[5:3], its bit mask is
  bit5 | bit9 | bit10;
- Group "ptp" has 3 pins in A0 as Z1, but its bitmask is changed to
  bit11 | bit12 | bit13.

Change-Id: Idabdb0578082e42e50f31342de3d3c2c9de750dd
Reviewed-on: http://vgitil04.il.marvell.com:8080/41970
Verified-Armada37x0: Wilson Ding <dingwei@marvell.com>
Tested-by: Wilson Ding <dingwei@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42775
Reviewed-by: Evan Wang <xswang@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Tested-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../pinctrl/marvell,armada-37xx-pinctrl.txt        |    6 +++++-
 drivers/pinctrl/mvebu/pinctrl-armada-37xx.c        |    5 +++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/Documentation/devicetree/bindings/pinctrl/marvell,armada-37xx-pinctrl.txt b/Documentation/devicetree/bindings/pinctrl/marvell,armada-37xx-pinctrl.txt
index 1acd001..e146985 100644
--- a/Documentation/devicetree/bindings/pinctrl/marvell,armada-37xx-pinctrl.txt
+++ b/Documentation/devicetree/bindings/pinctrl/marvell,armada-37xx-pinctrl.txt
@@ -120,9 +120,13 @@ group rgmii
  - functions mii, gpio
 
 group pcie1
- - pins 39-40
+ - pins 39-41
  - functions pcie, gpio
 
+group smi
+ - pins 54-55
+ - functions smi, gpio
+
 group ptp
  - pins 56-58
  - functions ptp, gpio
diff --git a/drivers/pinctrl/mvebu/pinctrl-armada-37xx.c b/drivers/pinctrl/mvebu/pinctrl-armada-37xx.c
index f025b4e..d8d2bf4 100644
--- a/drivers/pinctrl/mvebu/pinctrl-armada-37xx.c
+++ b/drivers/pinctrl/mvebu/pinctrl-armada-37xx.c
@@ -182,8 +182,9 @@ struct armada_37xx_pinctrl {
 	PIN_GRP_GPIO("usb2_drvvbus1", 1, 1, BIT(1), "drvbus"),
 	PIN_GRP_GPIO("sdio_sb", 24, 6, BIT(2), "sdio"),
 	PIN_GRP_GPIO("rgmii", 6, 12, BIT(3), "mii"),
-	PIN_GRP_GPIO("pcie1", 3, 2, BIT(4), "pcie"),
-	PIN_GRP_GPIO("ptp", 20, 3, BIT(5), "ptp"),
+	PIN_GRP_GPIO("smi", 18, 2, BIT(4), "smi"),
+	PIN_GRP_GPIO("pcie1", 3, 3, BIT(5) | BIT(9) | BIT(10), "pcie"),
+	PIN_GRP_GPIO("ptp", 20, 3, BIT(11) | BIT(12) | BIT(13), "ptp"),
 	PIN_GRP("ptp_clk", 21, 1, BIT(6), "ptp", "mii"),
 	PIN_GRP("ptp_trig", 22, 1, BIT(7), "ptp", "mii"),
 	PIN_GRP_GPIO_3("mii_col", 23, 1, BIT(8) | BIT(14), 0, BIT(8), BIT(14),
-- 
1.7.9.5

